gsoc/models.py

```diff
@@ -0,0 +1,204 @@
+    slot_request = models.BooleanField()
+    slot_allocation_grace_period = models.BooleanField()
+    project_acceptance = models.BooleanField()
+    project_decisions_finalization = models.BooleanField()
+    first_work_period = models.BooleanField()
+    first_evaluations = models.BooleanField()
+    second_work_period = models.BooleanField()
+    second_evaluations = models.BooleanField()
+    third_work_period = models.BooleanField()
+    final_week = models.BooleanField()
+    final_evaluations_mentor = models.BooleanField()
+    post_program = models.BooleanField()
+    org_signup_open = models.OneToOneField(GsocEvent, related_name='org_signup_open')
+    org_approval = models.OneToOneField(GsocEvent, related_name='org_approval')
+    orgs_published = models.OneToOneField(GsocEvent, related_name='org_published')
+    student_signup_open = models.OneToOneField(GsocEvent, related_name='student_signup_open')
+    slot_request = models.OneToOneField(GsocEvent, related_name='slot_request')
+    slot_allocation_grace_period = models.OneToOneField(GsocEvent, related_name='slot_allocation_grace_period')
+    project_acceptance = models.OneToOneField(GsocEvent, related_name='project_acceptance')
+    project_decisions_finalization = models.OneToOneField(GsocEvent, related_name='project_decisions_finalization')
+    first_work_period = models.OneToOneField(GsocEvent, related_name='first_work_period')
+    first_evaluations = models.OneToOneField(GsocEvent, related_name='first_evaluations')
+    second_work_period = models.OneToOneField(GsocEvent, related_name='second_work_period')
+    second_evaluations = models.OneToOneField(GsocEvent, related_name='second_evaluations')
+    third_work_period = models.OneToOneField(GsocEvent, related_name='third_work_period')
+    final_week = models.OneToOneField(GsocEvent, related_name='final_week')
+    final_evaluations_mentor = models.OneToOneField(GsocEvent, related_name='final_evaluations_mentor')
+    post_program = models.OneToOneField(GsocEvent, related_name='post_program')
+    community_bonding_period = models.OneToOneField(GsocEvent, related_name='community_bonding_period')
+    coding_period = models.OneToOneField(GsocEvent, related_name='coding_period')
+    results_announced = models.DateField()
+    tax_forms_deadline = models.DateField()
+    first_payments = models.DateField()
+    second_payments = models.DateField()
+    final_payments = models.DateField()
+    first_evaluations_finalized = models.DateField()
+    second_evaluations_finalized = models.DateField()
+    final_evaluations_finalized = models.DateField()
+    projects_accepted_count = models.IntegerField()
+    org_accepted_count = models.IntegerField()
+    slot_allocation_finalized = models.DateField()
+    second_enrollment_form_upload_deadline = models.DateField()
+    gsoc = models.ForeignKey(Gsoc, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return str(self.year)
+
+
+class Feature(models.Model):
```

Comment:
The `Feature` set are defined as part of the `Program` ; 1-1 relationship by the look of things.  Needs a foriegn key here.

As such, it should be called `Features` , and could even be inlined into the `Program`.

The alternative is to define a `Feature` which is a name and a bool , and then `Program` would have an attribute `features`.

Reply:
rename to `Features`

Reply:
@jayvdb, Django will automatically add a 's' in the of the model names.